api4/usage_test.go

```diff
@@ -24,11 +26,21 @@ func TestGetPostsUsage(t *testing.T) {
 	})
 
 	t.Run("good request returns response", func(t *testing.T) {
-		// Following calls create a total of 15 posts
 		th := Setup(t).InitBasic()
 		defer th.TearDown()
-		th.CreatePost()
-		th.CreatePost()
+
+		for i := 0; i < 14; i++ {
+			th.CreatePost()
+		}
+
+		total, err := th.Server.Store.Post().AnalyticsPostCount(&model.PostCountOptions{ExcludeDeleted: true})
+		require.NoError(t, err)
+		usersOnly, err := th.Server.Store.Post().AnalyticsPostCount(&model.PostCountOptions{ExcludeDeleted: true, UsersPostsOnly: true})
+		require.NoError(t, err)
+
+		require.GreaterOrEqual(t, usersOnly, int64(14))
+		require.LessOrEqual(t, usersOnly, int64(20))
+		require.GreaterOrEqual(t, total, usersOnly)
```

Comment:
This is fine, but I was suggesting we do something like:

```Go
count1, err := th.Server.Store.Post().AnalyticsPostCount(&model.PostCountOptions{ExcludeDeleted: true})

newPostCount := 14
for i := 0; i < newPostCount; i++ {
    th.CreatePost()
}

count2, err := th.Server.Store.Post().AnalyticsPostCount(&model.PostCountOptions{ExcludeDeleted: true})

require.Equal(t, newPostCount, int64(count2 - count1)
```


We don't have to test the `UsersPostOnly` option again here, it's already tested in the store.

Reply:
I'm not sure, if I follow correctly, but the code you mentioned, seems to be validating the `CreatePost` functionality.

> We don't have to test the UsersPostOnly option again here, it's already tested in the store.

Yes, you're right, I think we can ignore the normal/good request flow case altogether, as it's already tested in the app & store. Especially, in this case where there is no business rule to be tested in the api func.